server/test/review.spec.js

```diff
@@ -0,0 +1,129 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const { server } = require("../app.js");
+const dotenv = require('dotenv')
+const request = require('supertest');
+
+dotenv.config();;
+
+chai.should();
+chai.use(chaiHttp);
+
+// valid credentials
+const userCredentials = {
+  email: 'jondoe@vmail.com',
+  password: '12345678',
+}
+
+const lorem = "Nori grape silver beet broccoli kombu beet greens fava bean potato quandong celery. Bunya nuts black-eyed pea prairie turnip leek lentil turnip greens parsnip. Sea lettuce lettuce water chestnut eggplant winter purslane fennel azuki bean earthnut pea sierra leone bologi leek soko chicory celtuce parsley jícama salsify.";
+
+//login the user before we run any tests
+var authenticatedUser = request.agent(server);
+before((done) => {
+  authenticatedUser
+    .post('/auth/login')
+    .send(userCredentials)
+    .end((err, res) => {
+      res.should.have.status(200);
+      done();
+    });
+});
+
+// Test adding a new review
+describe("/POST review", () => {
+  it("it should return 201 status the added review", (done) => {
+    // valid userId for whome review is added
+    const uid = "612401d05be2e31e30a8d65e";
```

Comment:
Is this a real uid or just a fake uid?

Reply:
It's a real one

Reply:
if it's real, it doesn't really follow best practices for unit tests. instead, your test should create a user with every test, and grab the uid.

Best practices for test is that it should not depend on a running server!

Reply:
Would it be better to just use a valid uid(since that's all the validation needs to pass)? Since I'm not checking for an review anyway, just that there is a reviews object array? Mainly because I would have to create a new user account everytime the test is run which could mean there will be lots of test accounts on server which I don't know whether it's preferable or not

Reply:
We can run code to clean up after tests are finished. We could actually potentially break our tests if we just use the same account for every test, if our tests change it can lead to problems. This is why we should be working on a fresh new account for every test, we can clean up the account after every test.

Reply:
I changed this in a new commit, now every test will create accounts for use for that test, I am also appending the emails and username with a set prefix and then some random number to make these accounts identifiable, also made some changes to test structure to accommodate it and changed it in every test to keep consistent with pattern(not sure if this part was necessary).